src/main/java/bio/terra/workspace/service/workspace/exceptions/StageDisabledException.java

```diff
@@ -0,0 +1,14 @@
+package bio.terra.workspace.service.workspace.exceptions;
+
+import bio.terra.workspace.common.exception.BadRequestException;
+import bio.terra.workspace.common.model.Workspace;
+
+/** Exception for when an operation on workspace is not allowed according to its stage. */
+public class StageDisabledException extends BadRequestException {
+  public StageDisabledException(Workspace workspace, String operationName) {
```

Comment:
Maybe pass workspace ID and stage instead of a whole workspace? I could go either way on this one

Reply:
Yeah me too. Switching it.